Fix warnings seen when running process-css-properties.py

weinig opened the pull request:
#### 4d082633f9d78ef54c78b7d7d68b4574a51e6a64
<pre>
Fix warnings seen when running process-css-properties.py
<a href="https://bugs.webkit.org/show_bug.cgi?id=288310">https://bugs.webkit.org/show_bug.cgi?id=288310</a>

Reviewed by Darin Adler.

- Adds missing non-standard &quot;intrinsic&quot; value to the values
  array for sizing properties.
- Adds missing non-standard &quot;auto&quot; and &quot;match-element&quot; values
  to the values array for view-transition-name.
- Disable warnings for properties that use a non-builtin
  reference term in their grammar. These grammars are
  incomplete from the generators perspective, so they must
  be treated like any other custom parsing.

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/process-css-properties.py:

Canonical link: <a href="https://commits.webkit.org/290923@main">https://commits.webkit.org/290923@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b5060522593deab39d9eb0b55567b49444d0842f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/91448 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/10980 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/501 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/96417 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/42137 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/93498 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/11357 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/19398 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/70219 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/27738 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/94449 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/8658 "Found 1 new test failure: fast/forms/ios/focus-input-in-fixed.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/82843 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/50545 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/8423 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/430 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/41306 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/78743 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/435 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/98420 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/18610 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/13716 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/79241 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/18865 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/78681 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/78445 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19408 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/22963 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/329 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/11743 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/18608 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/23884 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/18318 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/21778 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/20084 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on Source/WebCore/css/CSSProperties.json:
```diff
@@ -2223,6 +2223,10 @@
                     "value": "contain",
                     "status": "unimplemented"
                 },
+                {
+                    "value": "intrinsic",
+                    "status": "non-standard"
+                },
```
Seeing this makes me realize that it's hard to spot patterns in these JSON format files because they are so vertical and wordy.

weinig replied:
Yeah, JSON would not be choice for this. 

The reason it is JSON still is that it is used by both the build and to drive https://webkit.org/css-status/. Whatever we change it to, we would need to either a) duplicate some of it in a JSON file, b) have some way to generate the JSON file from the new format for the server, c) make the new format something that can be parsed in the browser as well.

For this particular thing, the values array, we ideally would just get rid of it, and use the grammars for this information. We run into the same problem as above here.

All solvable.

weinig committed:
Fix warnings seen when running process-css-properties.py
https://bugs.webkit.org/show_bug.cgi?id=288310

Reviewed by Darin Adler.

- Adds missing non-standard "intrinsic" value to the values
  array for sizing properties.
- Adds missing non-standard "auto" and "match-element" values
  to the values array for view-transition-name.
- Disable warnings for properties that use a non-builtin
  reference term in their grammar. These grammars are
  incomplete from the generators perspective, so they must
  be treated like any other custom parsing.

* Source/WebCore/css/CSSProperties.json:
* Source/WebCore/css/process-css-properties.py:

Canonical link: https://commits.webkit.org/290923@main
